### PR TITLE
avoid iteration in template

### PR DESCRIPTION
**Pull request by mariaschuld**
**Context:** During the templates re-design process @josh146 noticed that some *layer* templates are not compatible with the TensorFlow interface. It turns out the fix is very easy.

*Note: It is a bit harder to fix the embedding templates, since their feature inputs are not necessarily converted by the qnode.*

**Description of the Change:** In the Layers() templates, replace

``
for w in weights:
``
with
``
for l in range(len(weights)):
``

**Benefits:** Makes templates TF compatible.

**Possible Drawbacks:** None.

### CODE REVIEW

**Comment by mariaschuld** [comment]
Note: I'm busy writing tests for interface integration in a separate PR, but used them to test this fix. 

The tests here only check if the update breaks any NumPy functionality.

File: pennylane/templates/layers.py [hunk 1]
```diff
@@ -116,8 +117,9 @@ def RandomLayers(weights, wires, ratio_imprim=0.3, imprimitive=CNOT, rotations=N
     if rotations is None:
         rotations = [RX, RY, RZ]
 
-    for layer_weights in weights:
-        RandomLayer(layer_weights, wires=wires, ratio_imprim=ratio_imprim, imprimitive=imprimitive, rotations=rotations,
+    n_layers = len(weights)
+    for l in range(n_layers):
```
Comment (co9olguy) [inline pennylane/templates/layers.py]:
```suggestion
    for l in range(len(weights)):
```

File: pennylane/templates/layers.py [hunk 2]
```diff
@@ -116,8 +117,9 @@ def RandomLayers(weights, wires, ratio_imprim=0.3, imprimitive=CNOT, rotations=N
     if rotations is None:
         rotations = [RX, RY, RZ]
 
-    for layer_weights in weights:
-        RandomLayer(layer_weights, wires=wires, ratio_imprim=ratio_imprim, imprimitive=imprimitive, rotations=rotations,
+    n_layers = len(weights)
```
Comment (co9olguy) [inline pennylane/templates/layers.py]:
```suggestion
```

File: pennylane/templates/layers.py [hunk 3]
```diff
@@ -49,8 +49,9 @@ def StronglyEntanglingLayers(weights, wires, ranges=None, imprimitive=CNOT):
     if ranges is None:
         ranges = [1]*len(weights)
 
-    for block_weights, block_range in zip(weights, ranges):
-        StronglyEntanglingLayer(block_weights, r=block_range, imprimitive=imprimitive, wires=wires)
+    n_layers = len(weights)
```
Comment (co9olguy) [inline pennylane/templates/layers.py]:
```suggestion
```

File: pennylane/templates/layers.py [hunk 4]
```diff
@@ -49,8 +49,9 @@ def StronglyEntanglingLayers(weights, wires, ranges=None, imprimitive=CNOT):
...
+    for l, block_range in zip(range(n_layers), ranges):
```
Comment (co9olguy) [inline pennylane/templates/layers.py]:
```suggestion
    for l, block_range in zip(range(len(weights)), ranges):
```

**Comment by co9olguy** [comment]
sorry, I ninja'd in some tiny code quality commits, but I guess you ran into merge conflicts since he were working in another branch based off this

**Comment by josh146** [comment]
Thanks @mariaschuld! Something to keep in mind while writing the integration tests: checking that the TensorFlow interface works for multiple templates in a QNode (i.e., amplitude embedding followed by multiple layers)

**Pull request merged**